pygmt/clib/session.py

```diff
@@ -1801,9 +1806,12 @@ def read_virtualfile(
         # _GMT_DATASET).
         if kind is None:  # Return the ctypes void pointer
             return pointer
-        if kind in ["image", "cube"]:
-            raise NotImplementedError(f"kind={kind} is not supported yet.")
-        dtype = {"dataset": _GMT_DATASET, "grid": _GMT_GRID}[kind]
+        dtype = {
+            "dataset": _GMT_DATASET,
+            "grid": _GMT_GRID,
+            "image": _GMT_IMAGE,
+            "cube": _GMT_CUBE,
+        }[kind]
```

Comment:
If we merge this PR for PyGMT v0.12.0, then `NotImplementedError` won't be raised right?

Reply:
Yes, but since it's a low-level API function, I guess users won't call it anyway and we maintainers should know what we're doing.

Reply:
But `kind='image'` and `king='cube'` is showing up in the API docs at https://pygmt-dev--3181.org.readthedocs.build/en/3181/api/generated/pygmt.clib.Session.virtualfile_out.html. Can we push this PR to v0.13.0? There's no real benefit to users in having this merged anyway without `GMT_IMAGE`/`GMT_CUBE` being wrapped right?

Reply:
Oh wait, that was the docs for `virtualfile_out`, I meant https://pygmt-dev--3181.org.readthedocs.build/en/3181/api/generated/pygmt.clib.Session.read_virtualfile.html

Reply:
I guess we can close this PR, since it's likely we will finish wrapping `GMT_IMAGE` first and then `GMT_CUBE` later, so we only need to deal with the conflicts in PR #3150.